apm-lambda-extension/main.go

```diff
@@ -71,35 +69,19 @@ func main() {
 	}
 	extension.Log.Debugf("Register response: %v", extension.PrettyPrint(res))
 
-	// Create a channel to buffer apm agent data
-	agentDataChannel := make(chan extension.AgentData, 100)
-
-	// Start http server to receive data from agent
-	if err = extension.StartHttpServer(ctx, agentDataChannel, config); err != nil {
+	// Init APM Server Transport struct and start http server to receive data from agent
+	apmServerTransport := extension.InitApmServerTransport(config)
+	agentDataServer, err := extension.StartHttpServer(ctx, apmServerTransport)
+	if err != nil {
```

Comment:
There might be a way to remove one of these two calls, in a similar fashion to what we did with `logsapi.Subscribe` and `logsTransport`.